### PR TITLE
add regexp to cmp matcher

### PR DESCRIPTION
**Pull request by arlimus**
i.e. `123 should { cmp /2+/ }`

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @arlimus

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: lib/matchers/matchers.rb [hunk 1]
```diff
@@ -248,8 +248,8 @@ def try_match(actual, op, expected) # rubocop:disable Metrics/CyclomaticComplexi
     # if actual and expected are strings
     if expected.is_a?(String) && actual.is_a?(String)
       return actual.casecmp(expected) == 0 if op == :==
-    elsif expected.is_a?(Regexp) && actual.is_a?(String)
-      return !actual.match(expected).nil?
+    elsif expected.is_a?(Regexp) && (actual.is_a?(String) || actual.is_a?(Integer))
```
Comment (alexpop) [inline lib/matchers/matchers.rb]: I made this change to support the Integer example in the PR request.

**Comment by alexpop** [comment]
I made a small change to support the Integer example in the PR request. It will also provide the `cmp` matching capabilities for this use-case: https://github.com/chef/inspec/issues/666 👿 

Like:

``` ruby
describe file('/etc/shadow') do
  its('mode') { should cmp /0(0|4)00/ }
end
```

**UPDATE:**
^ because mode returns Integer at the moment, the regex will have to be adapted for Integers, like:

``` ruby
describe file('/etc/shadow') do
  its('mode') { should cmp /^(256|0)$/ }
end
```

or converted to octal string:

```
describe file('/etc/shadow').mode.to_s(8) do
  it { should cmp /(0|4)00/ }
end
```

or for a way to get the file mode in octal.

**Comment by alexpop** [comment]
While updating the integration tests for `cmp`, I've noticed that there were a bunch of other tests failing:

For example:

``` bash
bundle exec kitchen verify default-centos-67
...
Finished in 1.43 seconds (files took 1.22 seconds to load)
200 examples, 20 failures, 11 pending

Failed examples:

rspec  # Apache Config /etc/httpd/conf/httpd.conf LogLevel should eq "warn"
rspec  # Apache Config /etc/httpd/conf/httpd.conf MaxKeepAliveRequests should eq "100"
rspec  #  should cmp 0
rspec  # SSH Configuration Port should eq "22"
rspec  # SSH Configuration Port should cmp "22"
rspec  # SSH Configuration Port should cmp 22
rspec  # SSH Configuration Port should cmp 22.0
rspec  # SSH Configuration Port should not cmp 22.1
rspec  # SSH Configuration LogLevel should eq "INFO"
rspec  # SSH Configuration LogLevel should cmp "INFO"
rspec  # SSH Configuration LogLevel should cmp "info"
rspec  # SSH Configuration LogLevel should cmp "InfO"
rspec  # /etc/group users should include "root"
rspec  # #<Inspec::Resources::EtcGroupView:0x007f90a621f6b8> users should include "root"
rspec  # File /tmp/file mode should cmp "0765"
rspec  # File /mnt/iso-disk should be mounted
rspec  # File /mnt/iso-disk should be mounted
rspec  # Mount /mnt/iso-disk options should eq ["ro"]
rspec  # PostgreSQL query: show ssl; output should eq "on"
rspec  # User root groups should include ["root"]
```

Are integration tests not run on PRs?

**Comment by srenatus** [comment]
@alexpop: currently, they require vagrant and are not run automatically at all. Improvements needed! :+1:

**Comment by arlimus** [comment]
@alexpop let's address the missing tests separately, we might need a workaround with travis here...

Everything else looks great :+1: Thank you for the improvements

**Comment by fcaviggia** [comment]
Awesome work! These are great improvements!

**Pull request merged**